executor/containerdexecutor/executor.go

```diff
@@ -329,5 +295,62 @@ func (w *containerdExecutor) Exec(ctx context.Context, id string, process execut
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	return taskProcess.Start(ctx)
+
+	return w.runProcess(ctx, taskProcess, process.Resize, nil)
+}
+
+func (w *containerdExecutor) runProcess(ctx context.Context, p containerd.Process, resize <-chan executor.WinSize, started func()) (err error) {
+	err = p.Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	if started != nil {
+		started()
+	}
+
+	statusCh, err := p.Wait(context.Background())
+	if err != nil {
+		return err
+	}
+
+	var cancel func()
+	ctxDone := ctx.Done()
+	for {
+		select {
+		case <-ctxDone:
+			ctxDone = nil
+			var killCtx context.Context
+			killCtx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+			p.Kill(killCtx, syscall.SIGKILL)
+		case size := <-resize:
+			err := p.Resize(ctx, size.Cols, size.Rows)
+			if err != nil {
+				cancel()
+				return err
+			}
+		case status := <-statusCh:
+			if cancel != nil {
+				cancel()
+			}
+			if status.ExitCode() != 0 {
+				var err error
+				if status.ExitCode() == containerd.UnknownExitStatus && status.Error() != nil {
+					err = errors.Wrap(status.Error(), "failure waiting for process")
+				} else {
+					err = &executor.ExitError{
+						ExitCode: status.ExitCode(),
+						Err:      status.Error(),
+					}
+				}
+				select {
+				case <-ctx.Done():
+					err = errors.Wrap(ctx.Err(), err.Error())
```

Comment:
This wrap is wrong way if we want to keep the exit code as typed. Maybe we can have a custom `Unwrap` so that both remain typed. Or just set `err.Err` to `context.Canceled` ?

Reply:
Yeah, I debated if the ctx.Err was more or less relevant than the Run error.  I think setting err.Err to ctx.Err likely makes the most sense here, I will update.